### DISCUSSION TITLE
get method type to boolean is resolved to unknown

### DISCUSSION THREAD

**Discussion by C0ZEN** [cyrilletuzi/angular-async-local-storage#302] [Q&A] [answered]
Hello,

Since the migration from Angular 8 to 9,  
I have a type problem here with `storageMap.get()`.  

When using a generic type as boolean, the overload should be:
```typescript
get<T = boolean>(key: string, schema: JSONSchemaBoolean): Observable<boolean | undefined>;
```

But in reality, the result is:
```typescript
get<T = unknown>(key: string, schema?: JSONSchema): Observable<unknown>;
```

You can checkout the [problem on StackBlitz](https://stackblitz.com/edit/angular-onxxkrangular-async-local-storage-get-type-bug?file=src%2Fapp%2Fapp.component.ts) on the app.component.ts.

**Comment by cyrilletuzi** [comment, marked as answer]
Hello, thanks for the precise issue and the stackblitz reproduction.

This is working as intended.

You're getting a value from client-side storage. So assuming you'll get what you previously stored is not safe: it could have been forged. So it's why the lib is returning `unknown` if you don't provide a JSON schema for *runtime* validation (validation can't be done at *compilation* time via a cast). You can see in the overload you provided that the second param is *required*.

So what you need to do is this:
```ts
this.storage.get('hello', { type: 'boolean' });
```

But this is not a change in v9 of this lib. It was a change in v7. So maybe something during Angular v8 to v9 migration made you aware of this, but this is not new.

More information about:
- [background of the change from `any` to `unknown` in v7](https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/MIGRATION_TO_V7.md)
- [why validation is done via JSON schema and not via a cast, and how to validate properly since v8+](https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/VALIDATION.md)

**Comment by C0ZEN** [comment]
Thank you for the information.  

And yes, Angular 9 uses now a more recent version of TypeScript.  
I guess than the type system checker is less permissive.
